src/reconcile/rec_visibility.c

```diff
@@ -667,44 +332,34 @@ __wt_rec_upd_select(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_INSERT *ins, v
         r->max_ts = max_ts;
 
     /*
-     * TODO WT-5209: Consider max_txn, max_ts from las to be aggregated into r->max_txn and
-     * r->max_ts, reconciliation should not write max that is less than in page_las.
+     * If the update we chose was a birthmark.
      */
-
-    /*
-     * If the update we chose was a birthmark, or we are doing update-restore and we skipped a
-     * birthmark, the original on-page value must be retained.
-     */
-    if (upd != NULL && (upd->type == WT_UPDATE_BIRTHMARK ||
-                         (F_ISSET(r, WT_REC_UPDATE_RESTORE) && skipped_birthmark))) {
+    if (upd != NULL && upd->type == WT_UPDATE_BIRTHMARK) {
         /*
          * Resolve the birthmark now regardless of whether the update being written to the data file
          * is the same as it was the previous reconciliation. Otherwise lookaside can end up with
          * two birthmark records in the same update chain.
          */
-        WT_ERR(__rec_append_orig_value(session, page, ins, ripcip, first_inmem_upd, vpack));
+        WT_RET(__rec_append_orig_value(session, page, ins, ripcip, first_upd, vpack));
         upd_select->upd = NULL;
-        orig_val_appended = true;
     }
 
-    /*
-     * Check if all updates on the page are visible, if not, it must stay dirty.
-     *
-     * Updates can be out of transaction ID order (but not out of timestamp order), so we track the
-     * maximum transaction ID and the newest update with a timestamp (if any).
-     */
-    all_stable = ((upd != NULL && upd->ext != 0) || upd == first_txn_upd) && !list_prepared &&
-      !list_uncommitted && __wt_txn_visible_all(session, max_txn, max_ts);
+    /* Check if all updates on the page are committed, if not, it must stay dirty */
+    all_committed = upd == first_txn_upd && __wt_txn_visible_all(session, max_txn, max_ts);
```

Comment:
I tried to remove __wt_txn_visible_all(session, max_txn, max_ts) but more tests failed. Do we have the assumption that the updates on a clean page must be all visible? I think in durable history, this is not true anymore. @bvpvamsikrishna